### PR TITLE
Fix /readyz to contain informer-sync

### PR DESCRIPTION
**Pull request by wojtek-t**
Ref https://github.com/kubernetes/kubernetes/issues/93599

```release-note
Fix kube-apiserver /readyz to contain "informer-sync" check ensuring that internal informers are synced.
```

/kind bug

### CODE REVIEW

**Comment by wojtek-t** [comment]
/hold

For @liggitt  to take a look

**Comment by wojtek-t** [comment]
/retest
[verify timeout]

File: test/e2e/apimachinery/health_handlers.go [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientset "k8s.io/client-go/kubernetes"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	"github.com/onsi/ginkgo"
+)
+
+var (
+	expectedHealthzVerbose = `[+]ping ok
```
Comment (liggitt) [inline test/e2e/apimachinery/health_handlers.go:33]: these are pretty fragile to apiserver configuration... for example, rbac might not be present in some clusters. also, other configurations (like KMS-based encryption add checks)

File: test/e2e/apimachinery/health_handlers.go [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+	if output := string(body); output != expectedOutput {
+		return fmt.Errorf("unexpected result: %s, expected %s", output, expectedOutput)
+	}
+	return nil
+}
+
+var _ = SIGDescribe("health handlers", func() {
+	f := framework.NewDefaultFramework("health")
+
+	ginkgo.It("/healthz should contain all checks", func() {
```
Comment (liggitt) [inline test/e2e/apimachinery/health_handlers.go:124]: three distinct tests seems like overkill... maybe one with three calls it verifies? remember there is per test overhead like namespace creation, etc

File: test/e2e/apimachinery/health_handlers.go (hunk 1 above)
Comment (wojtek-t) [reply test/e2e/apimachinery/health_handlers.go]: Changed.

Please let me know what checks should be removed from required ones.

File: test/e2e/apimachinery/health_handlers.go (hunk 2 above)
Comment (wojtek-t) [reply test/e2e/apimachinery/health_handlers.go]: Done.

File: test/e2e/apimachinery/health_handlers.go [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+		"[+]ping ok",
+		"[+]log ok",
+		"[+]etcd ok",
+		"[+]poststarthook/start-kube-apiserver-admission-initializer ok",
+		"[+]poststarthook/generic-apiserver-start-informers ok",
+		"[+]poststarthook/start-apiextensions-informers ok",
+		"[+]poststarthook/start-apiextensions-controllers ok",
+		"[+]poststarthook/crd-informer-synced ok",
+		"[+]poststarthook/bootstrap-controller ok",
+		"[+]poststarthook/rbac/bootstrap-roles ok",
```
Comment (liggitt) [inline test/e2e/apimachinery/health_handlers.go:45]: rbac is optional

File: test/e2e/apimachinery/health_handlers.go [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+		"[+]poststarthook/scheduling/bootstrap-system-priority-classes ok",
+		"[+]poststarthook/start-cluster-authentication-info-controller ok",
+		"[+]poststarthook/aggregator-reload-proxy-client-cert ok",
```
Comment (liggitt) [inline test/e2e/apimachinery/health_handlers.go:48]: this is optional based on API server flags

File: test/e2e/apimachinery/health_handlers.go [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+	body, err := result.Raw()
+	if err != nil {
+		return err
+	}
+	checks := sets.NewString(strings.Split(string(body), "\n")...)
+	for _, check := range requiredChecks {
+		if !checks.Has(check) {
+			return fmt.Errorf("required %s check: %s not present: %s", path, check, string(body))
+		}
+	}
```
Comment (liggitt) [inline test/e2e/apimachinery/health_handlers.go:121]: nit: testing if `requiredChecks.Difference(checks)` is non-empty and logging all missing required checks avoids iterating on CI adding a single string each time

**Comment by wojtek-t** [comment]
@liggitt - thanks! PTAL

**Comment by wojtek-t** [comment]
/retest
test-infra issues

**Pull request merged**